src/fusion-order/fusion-order.ts

```diff
@@ -1,37 +1,100 @@
-import {SETTLEMENT_CONTRACT_ADDRESS_MAP} from '../constants'
-import {AuctionSalt} from '../auction-salt'
-import {AuctionSuffix} from '../auction-suffix'
-import {
-    LimitOrder,
-    InteractionsData,
-    LimitOrderV3Struct,
-    OrderInfoDataFusion
-} from '../limit-order'
+import {SETTLEMENT_EXTENSION_ADDRESS_MAP} from '../constants'
+import {AuctionDetails} from '../auction-details'
+import {PostInteractionData} from '../post-interaction-data'
+import {LimitOrder, OrderInfoDataFusion, MakerTraits} from '../limit-order'
+import {FusionExtension} from './fusion-extension'
+import assert from 'assert'
+import {AuctionCalculator} from '../auction-calculator'
 
 export class FusionOrder extends LimitOrder {
+    public readonly extension: FusionExtension
+
     constructor(
         orderInfo: OrderInfoDataFusion,
-        private readonly auction: AuctionSalt,
-        private readonly auctionSuffix: AuctionSuffix,
-        interactions?: InteractionsData
+        auctionDetails: AuctionDetails,
+        postInteractionData: PostInteractionData,
+        extra: {
+            unwrapWETH?: boolean
+            /**
+             * Required if `allowPartialFills` is false
+             */
+            nonce?: bigint
+            permit?: string
+            /**
+             * Default is true
+             */
+            allowPartialFills?: boolean
+            /**
+             * Default deadline is 2m
+             */
+            deadline?: bigint
+        } = {}
     ) {
+        const makerTraits = MakerTraits.default()
+            .withExpiration(
+                extra.deadline || auctionDetails.auctionStartTime + 120n
+            )
+            .allowMultipleFills()
+
+        const allowPartialFills = extra.allowPartialFills ?? true
+
+        if (allowPartialFills) {
+            makerTraits.allowPartialFills()
+        } else {
+            makerTraits.disablePartialFills()
+
+            assert(
+                extra.nonce !== undefined,
+                'Nonce required, when partial fills disabled'
+            )
```

Comment:
Actually I think that better make the user to generate it, because otherwise in case of collision it will be hard to understand the problem for user

We can provide helper like `randNonce` in sdk to simplify usage